### PR TITLE
codeintel: Prefer higher-star count definition repos

### PR DESCRIPTION
**Pull request by efritz**
When a SCIP index defines a symbol, we write that index to a metadata table putting it into candidacy for cross-repo j2d on that symbol. We don't have many protections on what indexes can define what symbols (as there are no intuitive rules to apply across all ecosystems). This means that some random repo on GitHub may squat symbol names that should be reserved by a more popular library.

This can happen for repos that embed other repos and get indexed, and will generally happen only on _megainstanaces_ such as sourcegraph.com that does not have a cohesive set of repositories (and owners aware of the Sourcegraph instance).

To get around this name-squatting, we can prefer higher-starred repositories over lower ones. This will generally mean that libraries with lots of uses will be "protected" by this ordering. This applies for standard libraries as well, when indexed from a repository (which is the case for Go).

Original debug thread on slack: https://sourcegraph.slack.com/archives/CHXHX7XAS/p1690492726116199

## Test plan

- [x] Existing unit tests.
- [ ] Hand test the query plan on sourcegraph.com prior to merging to ensure we don't regress in xrepo performance.

### CODE REVIEW

**Review by Numbers88s — APPROVED** [review]

File: internal/codeintel/uploads/internal/store/uploads.go [hunk 1]
```diff
@@ -585,6 +585,7 @@ ranked_uploads AS (
 		-- within a repository by commit date. We'll choose the oldest commit
 		-- date as the canonical choice used to resolve the current definitions
 		-- request.
+		repo.stars,
```
Comment (keynmol) [inline internal/codeintel/uploads/internal/store/uploads.go:588]: does the comment above need an update?

Comment (efritz) [reply internal/codeintel/uploads/internal/store/uploads.go:588]: Updating

Comment (efritz) [reply internal/codeintel/uploads/internal/store/uploads.go:592]: [237771f](https://github.com/sourcegraph/sourcegraph/pull/55773/commits/237771fd91c51b89e848a0f69887dc039bca7524)

**Comment by varungandhi-src** [comment]
Closing as no test is present, can revive later if needed.

**Pull request closed without merging**